sparse_ho/tests/test_grid_search.py

```diff
@@ -17,21 +21,82 @@
 X, y, _ = make_correlated_data(
     n_samples, n_features, corr=corr, snr=snr, random_state=42)
 sigma_star = 0.1
+y = np.sign(y)
+
 
 idx_train = np.arange(0, 50)
 idx_val = np.arange(50, 100)
 
 alpha_max = np.max(np.abs(X[idx_train, :].T @ y[idx_train])) / len(idx_train)
 
 log_alphas = np.log(alpha_max * np.geomspace(1, 0.1))
-
 log_alpha_max = np.log(alpha_max)
 log_alpha_min = np.log(0.0001 * alpha_max)
 
 estimator = linear_model.Lasso(
-    fit_intercept=False, max_iter=1000, warm_start=True)
-
+    fit_intercept=False, max_iter=10000, warm_start=True)
+model = Lasso(estimator=estimator)
+
+tol = 1e-8
+
+# Set models to be tested
+models = {}
+models["lasso"] = Lasso(estimator=None)
+# models["enet"] = ElasticNet(estimator=None)
+# models["logreg"] = SparseLogreg(estimator=None)
+
+models["lasso_custom"] = Lasso(estimator=celer.Lasso(
+    warm_start=True, fit_intercept=False))
+# custom_models["enet"] = ElasticNet(
+#     estimator=linear_model.ElasticNet(warm_start=True, fit_intercept=False))
+# models["logreg_custom"] = SparseLogreg(
+#     estimator=celer.LogisticRegression(warm_start=True, fit_intercept=False))
```

Comment:
we have a lot of code duplication across the tests.

typically things you put in the header of your tests could be factorized in a file `sparse_ho/tests/common.py`

Reply:
done

Reply:
done